impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -188,10 +194,34 @@ public boolean replace(K key, V oldValue, V newValue) throws CacheAccessExceptio
 
   @Override
   public void clear() throws CacheAccessException {
+    boolean interrupted = false;
+    while(!cachingTierRef.compareAndSet(realCachingTier, noopCachingTier)) {
+      synchronized (noopCachingTier) {
+        if(cachingTierRef.get() == noopCachingTier) {
+          try {
+            noopCachingTier.wait();
+          } catch (InterruptedException e) {
+            interrupted = true;
+          }
+        }
+      }
+    }
+    if(interrupted) {
```

Comment:
A try finally might be nice although I admit it's not strictly necessary... also do we want to think about pulling this inside the loop?  That might help performance?